ml/asr/model.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+#include <string>
+#include <vector>
+#include <memory>
+#include "absl/status/statusor.h"
+#include "vosk_api.h"
+
+namespace aikit::ml {
+
+struct ASRResult {
+    std::string text;
+    std::vector<float> spk_embedding;
+};
+
+
+class ASRModel {
+public:
+  ASRModel(const std::string& model_path, const std::string& spk_model_path, size_t sample_rate = 16000);
+  ~ASRModel();
+
+  ASRModel(ASRModel&& other) noexcept;
+
+  ASRModel& operator=(ASRModel&& other) noexcept;
+
+  ASRModel(const ASRModel&) = delete;
+  ASRModel& operator=(const ASRModel&) = delete;
+
+  absl::StatusOr<ASRResult> operator()(const std::vector<float>& audio_buffer);
+
+public:
+  static constexpr size_t sample_rate = 16000;
+  static constexpr size_t channels = 1;
```

Comment:
To make user API clearer, let's remove these two variables (we donot use them anyway)